source/Mask.cpp

```diff
@@ -276,6 +276,16 @@ namespace {
 			radius = max(radius, p.LengthSquared());
 		return sqrt(radius);
 	}
+
+	double nearestPointOfSegmentToOrigin(Point start, Point end)
+	{
+		// https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line#Line_defined_by_two_points
+		// simplified for x0 = 0 and y0 = 0;
+		const double num = abs((end.X() - start.X()) * start.Y() - start.X() * (end.Y() - start.Y()));
```

Comment:
This can be simplified to the equivalent `abs(start.Cross(start - end))` I think.

Reply:
Thank's, I don't know much about the codebase... 

I will check but for this to work is important that the check is against the _segment_, not against the _line_

Reply:
No: it doesn't work :P Indeed after changing the code, I tried adding these lines just after the first check:
```
	double min_distance = min(distance, vA.Length());
	if (min_distance > nearestPointOfSegmentToOrigin(sA, vA)) throw 10;
```
And I obtained `terminate called after throwing an instance of 'int'` which means
that the implementation you suggested returned a distance lower than that of the nearest vertex.

Probably this deserve a comment...

Reply:
Ah I think I understand what your goal is now. I'm also very new to this code base, but this idea sounds very worthwhile as an extra quick check indeed! Very nice find!

However your code does also compute the distance to the line, not to the segment. E.g.:
```
	Point start(1000., 5.);
	Point end(2000., 5.);
	printf("%f\n", nearestPointOfSegmentToOrigin(start, end));
```
prints 5.

To get the distance to the segment I don't think you can easily do that without at least a few branches to detect the various cases. Working out the linear algebra, I think I would probably compute `start.Dot(start - end) / start.DistanceSquared(end)`. If that is between 0 and 1 then the point on the line closest to the origin in fact lies on the segment (so you can return the value you already computed), if it is below 0 then the closest point is start, and if it is above 1 then the closest point is end. (This needs verification and testing though of course...)

As an extra comment it would also be good to detect if start.DistanceSquared(end) is zero to avoid divisions by zero.

Reply:
Checked: you're right, `nearestPointOfSegmentToOrigin` is bugged even on mine (...and I don't understand how it ran: I did the same I suggested to you :dizzy_face: ...)

I will try as per your last comment.

Many thank's, I felt this couldn't be so easy!

Reply:
Done, should work now.